src/main/java/com/ipiecoles/java/eval/mdd050/service/ArtistService.java

```diff
@@ -35,20 +38,38 @@ public Artist findById(Long id) {
         }
         return artist.get();
     }
-
+    
+    public boolean findByName(String name) {
+    	Optional<Artist> artist = this.artistRepository.findByName(name);
+        if(!artist.isPresent()){
+            return false;
+        }
+        return true;
+    }
     public Long countAllArtists() {
         return artistRepository.count();
     }
-
+    
+    @Valid
     public Artist creerArtiste(Artist artist) {
+    	if (findByName(artist.getName())) {
+    		throw new DataIntegrityViolationException("Erreur : Artiste existant déjà en base de donnée");
+    	}
         return artistRepository.save(artist);
     }
 
     public void deleteArtist(Long id) {
+    	if (findById(id).equals(null)) {
+    		throw new DataIntegrityViolationException("L'artiste n'existe pas et ne peut donc être supprimé (logique).");
+    	}
```

Comment:
Dans ton cas, findById ne peut jamais renvoyer null. Car ta méthode lève une exception. On ne fait pas de equals non plus avec null, on utilise ==. Il te suffit d'appeler la méthode qui lèvera l'exception si l'artiste n'est pas trouvé. Si tu veux customiser le message, utilise un try catch.
```suggestion
    	findById(id);
```